Sources/Sentry/SentryHttpInterceptor.m

```diff
@@ -0,0 +1,112 @@
+#import "SentryHttpInterceptor+Private.h"
+#import "SentryHub+Private.h"
+#import "SentrySDK+Private.h"
+#import "SentryScope+Private.h"
+#import "SentryTraceHeader.h"
+
+@interface
+SentryHttpInterceptor () <NSURLSessionDelegate>
+
+@property (nonatomic, strong) NSURLSession *session;
+
++ (void)configureSessionConfiguration:(NSURLSessionConfiguration *)configuration;
+
+@end
+
+@implementation SentryHttpInterceptor
+
++ (void)configureSessionConfiguration:(NSURLSessionConfiguration *)configuration
+{
+    if (configuration == nil)
+        return;
+
+    NSMutableArray *protocolClasses = configuration.protocolClasses != nil
+        ? [NSMutableArray arrayWithArray:[configuration protocolClasses]]
+        : [[NSMutableArray alloc] init];
+
+    if (![protocolClasses containsObject:[self class]]) {
+        [protocolClasses insertObject:[self class] atIndex:0];
+    }
+
+    configuration.protocolClasses = protocolClasses;
+}
+
++ (BOOL)canInitWithRequest:(NSURLRequest *)request
+{
+    // Intercept the request if it is a http/https request
+    // not targeting Sentry and there is transaction in the scope.
+    NSNumber *intercepted = [NSURLProtocol propertyForKey:SENTRY_INTERCEPTED_REQUEST
```

Comment:
`m`: Why don't we just check if the request has already a `SENTRY_TRACE_HEADER`?

Reply:
Good point. I'm using an old solution that I use in another context.
I'll fix this.

Reply:
The request may not have the header in the unlikely case where there is no span in the scope when ```canonicalRequestForRequest``` is called.

Reply:
I get it. Thanks for explaining. Can we please add this info as a comment?